src/createUnistyles.ts

```diff
@@ -57,7 +58,7 @@ export const createUnistyles = <B extends Breakpoints, T = {}>(breakpoints: B) =
                         ...acc,
                         [key]: parseStyle<ST, B>(style, breakpoint, screenSize, sortedBreakpointEntries)
                     })
-                }, {} as ST)
+                }, {} as ST), [breakpoint, screenSize, sortedBreakpointEntries])
```

Comment:
you don't need to pass `sortedBreakpointEntries` to the deps. It's computed only once, before uses `useStyles`. Value will never change.

Reply:
`theme` needs to be passed to the deps array too otherwise theme won't be changed after user changes it

Reply:
Ah, of course. It looks like `parsedStyles` is actually what should be passed in here, that's the actual dependency which relies on `theme` and `stylesheet`.

Reply:
yeah you're 💯 right! Thanks @levibuzolic